streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -353,11 +371,19 @@ public void setHeaders(final Headers headers) {
      * The context exposes this metadata for use in the processor. Normally, they are set by the Kafka Streams framework,
      * but for the purpose of driving unit tests, you can set it directly. Setting this attribute doesn't affect the others.
      *
-     * @param timestamp A record timestamp
+     * @param recordTimestamp A record timestamp
      */
     @SuppressWarnings({"WeakerAccess", "unused"})
-    public void setTimestamp(final long timestamp) {
-        this.timestamp = timestamp;
+    public void setRecordTimestamp(final long recordTimestamp) {
+        this.recordTimestamp = recordTimestamp;
+    }
+
+    public void setCurrentSystemTimeMs(final long currentSystemTimeMs) {
+        this.currentSystemTimeMs = currentSystemTimeMs;
+    }
+
+    public void setCurrentStreamTimeMs(final long currentStreamTimeMs) {
+        this.currentStreamTimeMs = currentStreamTimeMs;
```

Comment:
This class is public API, so we cannot remove `setTimestamp` but can only deprecate it.

We also need to update the KIP to mention the deprecation and the newly added methods.

Reply:
@mjsax added the method setTimestamp back which sets record timestamp and added Deprecated annotation.